### PR TITLE
[bitnami/mongodb-sharded] Add again removed affinity option (dataNodeLoopId)

### LINKED ISSUE #4918: [mongodb-sharded] shard index as a label
Somehow this change is gone.

https://github.com/bitnami/charts/pull/2166

https://github.com/bitnami/charts/blob/60525899c60309f42520ce0b4f3f4c3b2f27a21e/bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml#L41

master

https://github.com/bitnami/charts/blob/master/bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml

This is the change removed this feature

https://github.com/bitnami/charts/pull/4494

### PR DESCRIPTION
**Pull request by carrodher**
**Description of the change**

During the affinity refactor (https://github.com/bitnami/charts/issues/4918), the part related to `dataNodeLoopid` was removed by mistake:
```diff
-      affinity: {{- include "mongodb-sharded.tplValue" ( dict "value" $.Values.shardsvr.dataNode.affinity "context" (set $ "dataNodeLoopId" $i) ) | nindent 8 }}
+      affinity: {{- include "common.tplvalues.render" (dict "value" $.Values.shardsvr.dataNode.affinity "context" $) | nindent 8 }}
```

**Applicable issues**

  - fixes #4918

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [n/a] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ spec:
       schedulerName: {{ $.Values.common.schedulerName | quote }}
       {{- end }}
       {{- if $.Values.shardsvr.dataNode.affinity }}
-      affinity: {{- include "common.tplvalues.render" (dict "value" $.Values.shardsvr.dataNode.affinity "context" $) | nindent 8 }}
+      affinity: {{- include "common.tplvalues.render" (dict "value" $.Values.shardsvr.dataNode.affinity "context" (set $ "dataNodeLoopId" $i)) | nindent 8 }}
```
Comment (juan131) [inline bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml:43]: This is useless the `.Values.shardsvr.dataNode.affinity` parameter has the block below in it:
```yaml
              matchLabels:
                shard: "{{ .dataNodeLoopId }}"
```
This should be reflected in the **values.yaml**, currently it doesn't say anything about this, see:
```yaml
    ## Affinity for pod assignment
    ## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
    ## Note: shardsvr.dataNode.podAffinityPreset, shardsvr.dataNode.podAntiAffinityPreset, and shardsvr.dataNode.nodeAffinityPreset will be ignored when it's set
    ##
    affinity: {}
```

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml (hunk 1 above)
Comment (carrodher) [reply bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml:43]:
```console
$ helm template . -s templates/shard/shard-data-statefulset.yaml -f values.yaml
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mongodb-sharded
        helm.sh/chart: mongodb-sharded-3.2.9
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: shardsvr
        shard: "0"
    spec:
      affinity:
        matchLabels:
          shard: '0'
...
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mongodb-sharded
        helm.sh/chart: mongodb-sharded-3.2.9
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: shardsvr
        shard: "1"
    spec:
      affinity:
        matchLabels:
          shard: '1'
```
just by setting
```yaml
affinity:
  matchLabels:
    shard: "{{ .dataNodeLoopId }}"
```
in the _values.yaml_

**Review by juan131 — APPROVED** [review]
LGTM

**Comment by maxisam** [comment]
I feel like it should work but it doesn't.  It renders to shard: "" , empty string.

```yml
    affinity:
      podAntiAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchLabels:
                shard: "{{ .dataNodeLoopId }}"
            topologyKey: kubernetes.io/hostname
```

**Comment by maxisam** [comment]
There are only 4 files changed. It should include chart.yml, the one with the chart version number

**Comment by carrodher** [comment]
Yes, those changes are included as part of the 3.2.10 version due to a new version appeared while this PR was in the test & release process. The new version is already released, you should be able to see the new changes after `helm repo update`.
```console
$ helm template . -s templates/shard/shard-data-statefulset.yaml -f values.yaml
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mongodb-sharded
        helm.sh/chart: mongodb-sharded-3.2.9
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: shardsvr
        shard: "0"
    spec:
      affinity:
        matchLabels:
          shard: '0'
...
  template:
    metadata:
      labels:
        app.kubernetes.io/name: mongodb-sharded
        helm.sh/chart: mongodb-sharded-3.2.9
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: shardsvr
        shard: "1"
    spec:
      affinity:
        matchLabels:
          shard: '1'
```
just by setting
```yaml
affinity:
  matchLabels:
    shard: "{{ .dataNodeLoopId }}"
```
in the values.yaml

**Pull request merged**